src/Nethermind/Nethermind.Synchronization/Synchronizer.cs

```diff
@@ -454,17 +258,141 @@ protected void WireFeedWithModeSelector<T>(ISyncFeed<T>? feed)
+                .Feed)
+            .Named<ISyncFeed<HeadersSyncBatch>>(nameof(HeadersSyncFeed));
+
+        ConfigureSnapComponent(builder);
+        ConfigureReceiptSyncComponent(builder);
+        ConfigureBodiesSyncComponent(builder);
+        ConfigureStateSyncComponent(builder);
+
+        builder
+            .RegisterNamedComponentInItsOwnLifetime<SyncFeedComponent<HeadersSyncBatch>>(nameof(HeadersSyncFeed), ConfigureFastHeader)
+            .RegisterNamedComponentInItsOwnLifetime<SyncFeedComponent<BlocksRequest>>(nameof(FastSyncFeed), ConfigureFastSync)
+            .RegisterNamedComponentInItsOwnLifetime<SyncFeedComponent<BlocksRequest>>(nameof(FullSyncFeed), ConfigureFullSync);
+    }
+
+    private void ConfigureFullSync(ContainerBuilder scopeConfig)
+    {
+        scopeConfig.AddScoped<ISyncFeed<BlocksRequest>, FullSyncFeed>();
+    }
+
+    private void ConfigureFastSync(ContainerBuilder scopeConfig)
+    {
+        if (syncConfig.FastSync)
+        {
+            scopeConfig.AddScoped<ISyncFeed<BlocksRequest>, FastSyncFeed>();
+        }
+        else
+        {
+            scopeConfig.AddScoped<ISyncFeed<BlocksRequest>, NoopSyncFeed<BlocksRequest>>();
+        }
+    }
+
+    private void ConfigureFastHeader(ContainerBuilder scopeConfig)
+    {
+        if (!syncConfig.FastSync || !syncConfig.DownloadHeadersInFastSync)
+        {
+            scopeConfig.AddScoped<ISyncFeed<HeadersSyncBatch>, NoopSyncFeed<HeadersSyncBatch>>();
+        }
+        else
+        {
+            scopeConfig.AddScoped<ISyncFeed<HeadersSyncBatch>, HeadersSyncFeed>();
+        }
+    }
+
+    private void ConfigureSnapComponent(ContainerBuilder serviceCollection)
+    {
+        serviceCollection
+            .AddSingleton<ProgressTracker>()
+            .AddSingleton<ISnapProvider, SnapProvider>();
+
+        ConfigureSingletonSyncFeed<SnapSyncBatch, SnapSyncFeed, SnapSyncDownloader, SnapSyncAllocationStrategyFactory>(serviceCollection);
+
+        if (!syncConfig.FastSync || !syncConfig.SnapSync)
+        {
+            serviceCollection.AddSingleton<ISyncFeed<SnapSyncBatch>, NoopSyncFeed<SnapSyncBatch>>();
+        }
+    }
+
+    private void ConfigureReceiptSyncComponent(ContainerBuilder serviceCollection)
+    {
+        ConfigureSingletonSyncFeed<ReceiptsSyncBatch, ReceiptsSyncFeed, ReceiptsSyncDispatcher, FastBlocksPeerAllocationStrategyFactory>(serviceCollection);
+
+        if (!syncConfig.FastSync || !syncConfig.DownloadHeadersInFastSync ||
+            !syncConfig.DownloadBodiesInFastSync ||
+            !syncConfig.DownloadReceiptsInFastSync)
+        {
+            serviceCollection.AddSingleton<ISyncFeed<ReceiptsSyncBatch>, NoopSyncFeed<ReceiptsSyncBatch>>();
+        }
+    }
+
+    private void ConfigureBodiesSyncComponent(ContainerBuilder serviceCollection)
+    {
+        ConfigureSingletonSyncFeed<BodiesSyncBatch, BodiesSyncFeed, BodiesSyncDownloader, FastBlocksPeerAllocationStrategyFactory>(serviceCollection);
```

Comment:
Do we still need to do this in case NoopSyncFeed is used?

Reply:
Yes, need the synchronizer also for FeedComponents. But its behind a `Lazy` so its never used. In any case, by default everything is lazy.